core/controller/src/main/scala/whisk/core/controller/Actions.scala

```diff
@@ -424,15 +425,22 @@ trait WhiskActionsApi extends WhiskCollectionAPI {
      * @return a promise that completes with (ActivationId, Some(WhiskActivation)) if blocking else (ActivationId, None)
      */
     private def postInvokeRequest(
-        user: Subject,
+        user: Identity,
         action: WhiskAction,
         env: Option[Parameters],
         payload: Option[JsObject],
         blocking: Boolean)(
             implicit transid: TransactionId): Future[(ActivationId, Option[WhiskActivation])] = {
         // merge package parameters with action (action parameters supersede), then merge in payload
         val args = { env map { _ ++ action.parameters } getOrElse action.parameters } merge payload
-        val message = Message(transid, s"/actions/invoke/${action.namespace}/${action.name}/${action.rev}", user, activationId.make(), args)
+        val message = Message(
+            transid,
+            FullyQualifiedEntityName(action.namespace, action.name, Some(action.rev())),
+            user.subject,
+            user.authkey,
+            activationId.make(),
+            user.subject.namespace,
```

Comment:
Should this rather be `user.namespace` (which also currently falls back to `subject` being the namespace anyway)? Would be more in line with where we want to be.